hirosystems/explorer
# chore: Dev Code review needed Adding playwright test for the explorer

This PR is the first addition of playwright test for the explorer

## Description

Please review the following test and add them into your current dev branch once this is done we can close this test PR


Below are the use cases we are running through for test automation, we will add test to this as we see fit. This serves as a baseline to add the Playwright test frame work to the repo and begin test automation.
Run the following on desktop in Google Chrome, FF and Safari

Test will run for Mobile on these browsers & Desktop on the following browsers Google chrome and FF



### Test scenario

- Go to the explorer
- Enter an address in the search bar 
- Verify a clickable link appears 
- Verify  the user can click the link and go to the expected page
- When the page loads get the STX balance and verify the number appears as expected.


### Test scenario 

- Go to the explorer
- Go to Transactions verify transactions are displaying and the page loads
- Click the latest transaction showing 
- Verify the page loads without an error opens 
- Verify the page doesn't time out
- Verify the page contains the following data correctly: 
_1. Summary_
_2. Amount_
_3. Sender address_
_4. Recipient_
_5. Transaction ID_

### Test scenario
- Go to the explorer
- Go to Blocks page 
- Verify the blocks are displaying and the page loads
- Click the latest block link and verify the page loads without error
- Verify the page doesn't time out
- Verify the page contains the following data correctly: 
_1. Summary_
_2. Hash_
_3. Block height_
_4. Mined_
_5. Transactions_


## Type of Change
- [X ] New feature

## Does this introduce a breaking change?
List the APIs or describe the functionality that this PR breaks.
Workarounds for or expected timeline for deprecation

## Are documentation updates required?
<!-- 
  DOCUMENTATION
Link to playwright docs:
 -https://codecept.io/playwright/#setup

## Checklist
- [ ] Tag 1 of @person1 or @person2
